pkg/tfbridge/provider.go

```diff
@@ -59,6 +59,114 @@ type Resource struct {
 	TFName string           // the Terraform resource name.
 }
 
+// collectImporterAttributes runs the Terraform Importer defined on the Resource for the given
+// resource ID, and adds any non-ID attributes to the inputs map.
+func (res *Resource) collectImporterAttributes(resourceID resource.ID, inputs map[string]string) error {
+	// There is nothing to do here if the resource doesn't have an importer defined in the
+	// Terraform schema.
+	if res.TF.Importer == nil {
+		return nil
+	}
+
+	glog.V(9).Infof("%s has TF Importer", res.TFName)
+
+	id := resourceID.String()
+
+	// Prepare a Terraform ResourceData for the importer
+	data := res.TF.Data(nil)
+	data.SetId(id)
+	data.SetType(res.TFName)
+
+	// Run the importer defined in the Terraform resource schema
+	results, err := res.TF.Importer.State(data, nil)
```

Comment:
Most of the other places we invoke resource providers we do so via the equivalent of `p.tf.ImportState`, which is effectively a wrapper over this.

I assume we're using the Importer directly here (and duplicating some of what `ImportState` does) because we want more control over some of the error cases?

Reply:
Right - we could use `ImportState` but we'd have to check for non-concrete-type errors when there is no importer (which is ok with us). In addition, the error would direct people to report bugs in Terraform instead of Pulumi, which is probably not ideal since they are really our bugs to deal with.